src/node_contractor.rs

```diff
@@ -146,12 +144,11 @@ mod tests {
         g.add_edge(1, 2, 1);
         g.add_edge(0, 3, 1);
         g.add_edge(3, 1, 1);
-        let _shortcuts = calc_shortcuts(&mut g, 1);
```

Comment:
Ok, lets look at this graph. We are contracting node 1 and there are two pairs of incoming/outgoing edges: 

p1: (0->1,1->2) weight(in+out)=11
p2: (3->1,1->2) weight(in+out)=2

During the contraction we take the first pair and do a witness search (with node 1 being the avoid node) and find that there is no other path as good as 0->1->2, so we will introduce the shortcut:

sc1:  0->2 (via 1), weight=11

Then we do the same for the second pair and we will add the shortcut:

sc2: 3->2 (via 1) weight=2

However, now we see that sc1 is not necessary, because instead of taking sc1 to get from 0 to 2 we can just as well go from 0 to 3 and then take sc2 from 3 to 2, which is cheaper. 

And now lets do the same, but this time we change the order of the edge pairs (we look at p2 first because it has lower weight). In this case we will add again sc2, but now look at the witness search for p1: We will find that 0->3->2 (going from 3->2 via sc2) is a witness path and therefore we will *not* introduce sc1 in this case. This is the reduction of shortcuts we are looking for (or at least this is what I meant when I added the todo in this test).
However, now there are two shortcuts and the first one turns out to be unecessary:

Reply:
in your case so still use the `avoid_node` function, which i do not use at all

Reply:
yes i agree, but the approach is still different